config/config_test.go

```diff
@@ -92,6 +95,72 @@ func Test_Kubeconfig_Env_Nonexisting(t *testing.T) {
 	}
 }
 
+// Test_Config_Migration tests the config dir migration from the old
+// to the new default location, including updating certificate paths
+// in the kubectl config file.
+func Test_Config_Migration(t *testing.T) {
+	defer viper.Reset()
+
+	// override standard paths for testing
+	dir := tempDir()
+	HomeDirPath = dir
+	DefaultConfigDirPath = path.Join(HomeDirPath, ".config", ProgramName)
+	LegacyConfigDirPath = path.Join(HomeDirPath, "."+ProgramName)
+
+	// create fake certs file
+	certsPath := path.Join(HomeDirPath, "."+ProgramName, "certs")
+	err := os.MkdirAll(certsPath, 0700)
+	if err != nil {
+		t.Error(err)
+	}
+	certFile, _ := os.Create(path.Join(certsPath, "test-ca.crt"))
+	certFile.Close()
+	certFile, _ = os.Create(path.Join(certsPath, "test-12345-client.crt"))
+	certFile.Close()
+	certFile, _ = os.Create(path.Join(certsPath, "test-12345-client.key"))
+	certFile.Close()
+
+	// add a test kubectl config file
+	kubeconfigpath := path.Join(dir, "tempkubeconfig")
+	KubeConfigPaths = []string{kubeconfigpath}
+	kubeConfig := []byte(`apiVersion: v1
+kind: Config
+preferences: {}
+current-context: g8s-system
+clusters:
+- cluster:
+    certificate-authority: ` + LegacyConfigDirPath + `/certs/test-ca.crt
+    server: https://api.n41en.g8s.fra-1.giantswarm.io
+  name: giantswarm-test
+users:
+- name: testuser
+  user:
+    client-certificate: ` + LegacyConfigDirPath + `/certs/test-12345-client.crt
+    client-key: ` + LegacyConfigDirPath + `/certs/test-12345-client.key
+contexts:
+- context:
+    cluster: giantswarm-test
+    user: testuser
+  name: giantswarm-test
+`)
+	fileErr := ioutil.WriteFile(kubeconfigpath, kubeConfig, 0700)
+	if fileErr != nil {
+		t.Error(fileErr)
+	}
+
+	t.Log("DefaultConfigDirPath:", DefaultConfigDirPath)
+	t.Log("LegacyConfigDirPath:", LegacyConfigDirPath)
+
+	migrateConfigDir()
+
+	// spit out config
+	newConf, confReadErr := ioutil.ReadFile(kubeconfigpath)
+	if confReadErr != nil {
+		t.Error(confReadErr)
+	}
+	t.Log(string(newConf))
```

Comment:
It will be output if the user starts tests in verbose mode (which btw Atom with go-plus does by default), so I still see a use for it.